### PR TITLE
doc: add Sidewalk doc sections

### PR DESCRIPTION
**Pull request by annwoj**
Added Sidewalk sections with documentation references. 
Included for both samples and protocols.

### CODE REVIEW

**Comment by annwoj** [comment]
@krlo @wojciechpluszynski Could you please take a look at this PR for Sidewalk and let me know if there's anything that should be rephrased/updated? Thank you!

**Review by wojciechpluszynski — COMMENTED** [review]
No issues from my side, Sidewalk disclaimer present and aligned with GitHub pages for Sidewalk.
References to main Sidewalk documentation OK.

**Review by RobertGalatNordic — APPROVED** [review]

**Review by ktaborowski — APPROVED** [review]
looks good to me

**Review by b-gent — APPROVED** [review]

File: doc/nrf/protocols/sidewalk/index.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   :depth: 2
+
+.. caution::
+   Sidewalk protocol is owned and licensed by Amazon.
+   Before continuing with the nRF Connect SDK for Sidewalk, you must familiarize yourself with the official `Amazon license`_, including all the applicable restrictions.
+   Please note, that redistributing Sidewalk under the nRF Connect SDK is prohibited.
+
+   Be aware, that by downloading and including the Sidewalk repository, you accept the above-mentioned license.
+
+   Due to these restrictions, Sidewalk cannot be automatically fetched through the nRF Connect SDK and will require performing a dedicated setup.
```
Comment (krlo) [inline doc/nrf/protocols/sidewalk/index.rst:17]: I would add a link pointing to the documentation describing beforementioned dedicated setup e.g.: "You can access documentation explaining how to add Amazon Sidewalk repository to nRF Connect SDK at http://..."

File: doc/nrf/samples/sidewalk.rst [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.. _sidewalk_samples:
+
+Sidewalk samples
```
Comment (krlo) [inline doc/nrf/samples/sidewalk.rst:3]: Change to Amazon Sidewalk samples

**Review by krlo — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/sidewalk/index.rst (hunk 1 above)
Comment (b-gent) [reply doc/nrf/protocols/sidewalk/index.rst]: added

**Pull request merged**